M2/cmake/check-libraries.cmake

```diff
@@ -352,6 +352,12 @@ else()
   unset(FACTORY_STREAMIO CACHE)
 endif()
 
+if(FLINT_FOUND)
+  check_include_files(flint/nmod.h HAVE_FLINT_NMOD_H)
```

Comment:
Ohh, we need something like this:
```cmake
  set(CMAKE_REQUIRED_INCLUDES "${FLINT_INCLUDE_DIR}")
```

Reply:
Otherwise it's just checking system directories.

Reply:
Gotcha -- I was wondering if we needed that or not!